server/src/test/java/e2e/StructDefLifecycleTest.java

```diff
@@ -0,0 +1,149 @@
+                .setAllowedUpdates(AllowedStructDefUpdateType.FULLY_COMPATIBLE_SCHEMA_UPDATES)
+                .setStructDef(InlineStructDef.newBuilder()
+                        .putFields(
+                                "model",
+                                StructFieldDef.newBuilder()
+                                        .setFieldType(
+                                                TypeDefinition.newBuilder().setType(VariableType.STR))
+                                        .build())
+                        .putFields(
+                                "year",
+                                StructFieldDef.newBuilder()
+                                        .setFieldType(
+                                                TypeDefinition.newBuilder().setType(VariableType.INT))
+                                        .setDefaultValue(
+                                                VariableValue.newBuilder().setInt(10))
+                                        .build())
+                        .putFields(
+                                "sold",
+                                StructFieldDef.newBuilder()
+                                        .setFieldType(
+                                                TypeDefinition.newBuilder().setType(VariableType.BOOL))
+                                        .setDefaultValue(
+                                                VariableValue.newBuilder().setBool(false))
+                                        .build()))
+                .build();
+
+        assertThat(client.putStructDef(updatedStructDef))
+                .extracting(item -> item.getId().getVersion())
+                .isEqualTo(1);
+    }
+
+    @Test
+    void shouldThrowErrorWhenPuttingIncompatibleStructDefEvolution() {
+        // TODO: Re-implement tests with StructDef builder helpers developed in following PR.
+        client.putStructDef(PutStructDefRequest.newBuilder()
+                .setName("car-3")
+                .setStructDef(InlineStructDef.newBuilder()
+                        .putFields(
+                                "model",
+                                StructFieldDef.newBuilder()
+                                        .setFieldType(
+                                                TypeDefinition.newBuilder().setType(VariableType.STR))
+                                        .build()))
+                .build());
+
+        PutStructDefRequest updatedStructDef = PutStructDefRequest.newBuilder()
+                .setName("car-3")
+                .setAllowedUpdates(AllowedStructDefUpdateType.FULLY_COMPATIBLE_SCHEMA_UPDATES)
+                .setStructDef(InlineStructDef.newBuilder()
+                        .putFields(
+                                "model",
+                                StructFieldDef.newBuilder()
+                                        .setFieldType(
+                                                TypeDefinition.newBuilder().setType(VariableType.STR))
```

Comment:
Beautiful test. Can we add another test case that does something like change `model` from `INT` to `STR`?